app/src/common/shared/com/igalia/wolvic/ui/widgets/WindowWidget.java

```diff
@@ -540,12 +540,7 @@ public void switchPanel(@Windows.PanelType int panelType) {
             hidePanel(true);
 
         } else {
-            if (mViewModel.getIsDownloadsVisible().getValue().get()) {
-                hideDownloadsPanel(false);
-            }
-            if (mViewModel.getIsAddonsVisible().getValue().get()) {
-                hideAddonsPanel(false);
-            }
+            hideAllPanel(false);
```

Comment:
Seems like these changes do not correspond to the commit title?

Also I wonder whether we could have the hideX -> hideAll transition in a single commit?

Reply:
I guess we can squash all these commits into one when we merge this PR. This PR is already big enough and it's painful to keep all the commits in order.

Reply:
Ok, I manually squashed all the commits into one from https://github.com/Igalia/wolvic/commit/28abf978cb20525eab7526a0aa5a7199b86a45e5 Otherwise it will have all kinds of conflicts, making address changes requests here almost impossible...